components/renku_data_services/user_preferences/blueprints.py

```diff
@@ -36,6 +37,8 @@ def post_pinned_projects(self) -> BlueprintFactoryResponse:
         @authenticate(self.authenticator)
         @validate(json=apispec.AddPinnedProject)
         async def _post(_: Request, body: apispec.AddPinnedProject, user: base_models.APIUser):
+            if "\u0000" in body.project_slug:
```

Comment:
**suggestion (non-blocking):** should we instead validate project slug to something like `a-zA-Z0-9-_`?